authz.go

```diff
@@ -225,6 +226,20 @@ func (a *authorizer) authorize(ctx context.Context, sr authz.SubjectAccessReview
 			break
 		}
 	}
+
+	// if cache and dry run are enabled, error need to be added to log if authorize result from cache and authorize result from zts / zms are different.
+	if a.AuthorizationConfig.Config.UseCache {
+		// decision is nil, meaning it is not authorized through cache, but zts/zms authorized
+		if decision == nil && granted {
+			log.Printf("There is a mismatch between cache result and zms result. Cache granted: unauthorized, Athenz granted: authorized for principal: %s on check: %s", principal, via)
+		}
+
+		// decision is not empty, but result from zts/zms is not authorize
+		if decision != nil && decision.status.Allowed && !granted {
```

Comment:
Instead of checking for explicit allowed from cache vs not granted from zts / zms calls, can we check if they're not equal and log it?

Reply:
This is because if zts / zms is returning true but our cache returned false then we also want to be aware of those cases.

Reply:
decision.status.Allowed will be always set to true if we get authorize from cache. this is for the second case when there's mismatch: authorized from cache and unauthorized from zts/zms.  not setting to equal we can be more explicit for this case.

Reply:
> This is because if zts / zms is returning true but our cache returned false then we also want to be aware of those cases.

I think this case is already covered in the first if statement.

Reply:
Ah okay got it. That clears it up. I thought the cache would return false instead of nil for those cases. This should be fine then.